Webtest.sh

```diff
@@ -61,11 +67,20 @@ docker exec $DOCKERNAME /usr/local/bin/fortio grpcping localhost
 # Do a grpcping to a scheme-prefixed destination. Fortio should append port number
 docker exec $DOCKERNAME /usr/local/bin/fortio grpcping https://fortio.istio.io
 docker exec $DOCKERNAME /usr/local/bin/fortio grpcping http://fortio.istio.io
+# Do a grpcping with -cert flag. Fortio should use valid cert.
+docker exec $DOCKERNAME /usr/local/bin/fortio grpcping -cacert $CERT fortio.istio.io:443
+docker exec $DOCKERNAME /usr/local/bin/fortio grpcping -cacert $CERT https://fortio.istio.io
 # Do a local grpcping. Fortio should append default grpc port number to destination
 docker exec $DOCKERNAME /usr/local/bin/fortio grpcping localhost
 # pprof should be there, no 404/error
 PPROF_URL="$BASE_URL/debug/pprof/heap?debug=1"
 $CURL $PPROF_URL | grep -i TotalAlloc # should find this in memory profile
+# switch to secure grpc mode and run grpcping and load test
+docker stop $DOCKERID
+docker rm $DOCKERNAME
+DOCKERID=$(docker run -d --ulimit nofile=$FILE_LIMIT --name $DOCKERNAME --mount type=bind,source=$PWD/ca.crt,target=$TESTCACERT --mount type=bind,source=$PWD/server.crt,target=$TESTCERT --mount type=bind,source=$PWD/server.key,target=$TESTKEY istio/fortio:webtest server -cert $TESTCERT -key $TESTKEY -loglevel $LOGLEVEL)
```

Comment:
In theory you'd mount the whole directory rather than 1 file at a time
and mount it to /etc/ssl/certs/fortio/ for instance to not shadow the existing /etc/ssl/certs which are needed for regular TLS tests
you could also then use the same image and just start another server on a different set of ports with the -cert / -key args, it'd be more efficient than a brand new docker run


**but** circleci doesn't support mounts because it's all remote docker and not local
https://circleci.com/docs/2.0/building-docker-images/#mounting-folders

you can try docker cp or we make a new layer or copy extra files/directory to  /etc/ssl/certs/fortio/ 
(that should be declared as a volume like /var/lib/istio/fortio is https://github.com/istio/fortio/blob/v0.8.1/Dockerfile#L41)

Reply:
thx for the ptr to the ci limitation. I have docker cp working for the cert/key files.